egs/mls_english/ASR/zipformer/train.py

```
@@ -0,0 +1,1396 @@
+            return False
+
+        # In pruned RNN-T, we require that T >= S
+        # where T is the number of feature frames after subsampling
+        # and S is the number of tokens in the utterance
+
+        # In ./zipformer.py, the conv module uses the following expression
+        # for subsampling
+        T = ((c.num_frames - 7) // 2 + 1) // 2
+        tokens = sp.encode(c.supervisions[0].text, out_type=str)
+
+        if T < len(tokens):
```

Ruff (0.12.2)

Yoda condition detected

Rewrite as `len(tokens) > T`

(SIM300)